File: packages/ruviz-web/examples/bar.ts
import { createPlot } from "../src/index.js";
import { categoricalSeries } from "./data.js";
import type { PlotExampleDefinition } from "./types.js";

export const example: PlotExampleDefinition = {
  slug: "bar",
  title: "Bar chart",
  summary: "Categorical metrics rendered as a bar chart.",
  category: "basic",
  createPlot() {
    const { categories, values } = categoricalSeries();
    return createPlot().sizePx(640, 320).title("Runtime Coverage").ylabel("score").bar({ categories, values });
  },
};

File: bindings/python/examples/bar.py
from __future__ import annotations

from _shared import ExampleMeta, base_plot, categorical_series, save_example

META = ExampleMeta(
    slug="bar",
    title="Bar chart",
    summary="Categorical metrics rendered as a bar chart.",
    section="Basic plots",
)


def build_plot():
    categories, values = categorical_series()
    return base_plot("Runtime Coverage").ylabel("score").bar(categories, values)


if __name__ == "__main__":
    save_example(META, build_plot())
